runtime/lua/vim/lsp.lua

```diff
@@ -960,27 +957,48 @@ function lsp.start_client(config)
       client.workspace_folders = workspace_folders
       -- TODO(mjlbach): Backwards compatibility, to be removed in 0.7
       client.workspaceFolders = client.workspace_folders
-      client.server_capabilities = assert(result.capabilities, "initialize result doesn't contain capabilities")
+
       -- These are the cleaned up capabilities we use for dynamically deciding
       -- when to send certain events to clients.
-      client.resolved_capabilities = protocol.resolve_capabilities(client.server_capabilities)
+      client.server_capabilities = assert(result.capabilities, "initialize result doesn't contain capabilities")
+      client.server_capabilities = protocol.resolve_capabilities(client.server_capabilities)
+
+      -- Deprecation wrapper: this will be removed in 0.8
+      local mt = {}
+      mt.__index = function(table, key)
+        if key == 'resolved_capabilities' then
+          vim.notify_once("[LSP] Accessing resolved_capabilities is deprecated, " ..
+          "update your plugins or configuration to access server_capabilities instead." ..
+          "The new key/value pairs in server_capabilities directly match those " ..
+          "defined in the language server protocol", vim.log.levels_warn)
```

Comment:
Although this warning might still be too long, readability-wise. Maybe a link or a `:h client.server_capabilities` is a good replacement for that last part?

```suggestion
          "update your plugins or configuration to use client.server_capabilities instead." ..
          "The new key/value pairs in client.server_capabilities directly match those " ..
          "defined in the language server protocol", vim.log.levels.WARN)
```